finesse/gui/measure_script/parse.py

```diff
@@ -0,0 +1,115 @@
+"""Code for parsing the YAML-formatted measure scripts."""
+import logging
+from dataclasses import dataclass
+from io import TextIOBase
+from pathlib import Path
+from typing import Any, Dict, Optional, Sequence, Union
+
+import yaml
+from pubsub import pub
+from PySide6.QtWidgets import QWidget
+from schema import And, Or, Schema, SchemaError
+
+from ...config import ANGLE_PRESETS
+from ..error_message import show_error_message
+
+
+@dataclass
+class Measurement:
+    """Represents a single measurement (i.e. angle + count)."""
+
+    angle: Union[str, float]
+    """Either an angle in degrees or the name of a preset angle."""
+
+    count: int
+    """The number of times to repeat the measurement at this position."""
+
+    def run(self) -> None:
+        """A placeholder function for recording multiple measurements."""
+        # Move the mirror to the correct location
+        pub.sendMessage("stepper.move", target=self.angle)
+
+        # Take the recordings
+        logging.info(f"Recording {self.count} measurements")
+
+
+class Script:
+    """Represents a measure script, including its file path and data."""
+
+    def __init__(
+        self, path: Path, count: int, sequence: Sequence[Dict[str, Any]]
+    ) -> None:
+        """Create a new Script.
+
+        Args:
+            path: The file path to this measure script
+            count: The number of times to repeat the sequence of measurements
+            sequence: Different measurements (i.e. angle + count) to record
+        """
+        self.path = path
+        self.count = count
+        self.sequence = [Measurement(**val) for val in sequence]
+
+    def run(self) -> None:
+        """Run this measure script."""
+        logging.info(f"Running {self.path}")
+        for i in range(self.count):
```

Comment:
I'd use [`tqdm`](https://pypi.org/project/tqdm/) here. Eventually, we will want a GUI-based progress bar somewhere that also allows to cancel the sequence, but that will be work for the future, when the basic stuff is up and running. Something like [QProgressDialog](https://doc.qt.io/qtforpython-5/PySide2/QtWidgets/QProgressDialog.html). Maybe open an issue about it, so we do not forget.

Reply:
This is a neat package. I knocked something together using it, but the problem is that it obscures the console output and atm that's the only acknowledgement we get that the mirror has been "moved" to a given position, so I think we should drop it for now.

I've opened an issue: #71